app/routes/circulars/circulars.lib.ts

```diff
@@ -143,3 +147,27 @@ export function formatAuthor({
   else if (!affiliation) return `${name} <${email}>`
   else return `${name} at ${affiliation} <${email}>`
 }
+
+export function parseEventFromSubject(value: string) {
+  const subjectMatchers: SubjectMatcher[] = [
+    [/GRB[.\s|_|-]*(\d{6}[a-zA-Z|.]\d*)/, ([, id]) => `GRB ${id}`],
+    [/SGR[.\s|_|-]*(\d{4}\.?\d*\+\d{4})/, ([, id]) => `SGR ${id}`],
+    [/IceCube[.\s|_|-]*(\d{6}[A-Z])/, ([, id]) => `IceCube ${id}`],
+    [/ZTF[.\s|_|-]*(\d{2}[a-zA-Z]*)/, ([, id]) => `ZTF ${id}`],
+    [/HAWC[.\s|_|-]*(\d{6}A)/, ([, id]) => `HAWC ${id}`],
+    [/LIGO\/Virgo[.\s|_|-]*(S\d{6}[a-zA-Z]+)/, ([, id]) => `LIGO/Virgo ${id}`],
+    [
+      /ANTARES[.\s|_|-]*(\d{6}[a-zA-Z])/,
+      ([, id]) => `ANTARES ${id}`.toUpperCase(),
+    ],
+    [
+      /Baksan\sNeutrino\sObservatory\sAlert[.\s|_|-]*(\d{6}.\d{2})/,
+      ([, id]) => `Baksan Neutrino Observatory Alert ${id}`,
+    ],
+  ]
```

Comment:
Move to module scope, so that we don't redefine all of these anonymous functions on every invocation.